presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -629,8 +630,15 @@ private PlanBuilder sort(PlanBuilder subPlan, List<SortItem> orderBy, Optional<S
 
         ImmutableList.Builder<Symbol> orderBySymbols = ImmutableList.builder();
         ImmutableMap.Builder<Symbol, SortOrder> orderings = ImmutableMap.builder();
+        Set<Symbol> orderBySymbolSet = new HashSet<Symbol>();
```

Comment:
Instead of keeping track of the symbols, I'd change the code to build a HashMap for `orderings` instead of Guava's ImmutableMap.builder.

One thing to be careful with is that if the sort order for a given column is different, the first one should win. For instance, in:

`SELECT a, a FROM t ORDER BY a ASC, a DESC`

`DESC` for the second occurrence is effectively a no-op.

I'd add a test for this while you're at it.